src/main/kotlin/com/hibob/academy/service/PetService.kt

```diff
@@ -0,0 +1,42 @@
+package com.hibob.academy.service
+
+import com.hibob.academy.dao.*
+import jakarta.inject.Inject
+import org.springframework.stereotype.Service
+
+@Service
+class PetService @Inject constructor(private val petDao: PetDao) {
+
+    fun createPet(pet: PetCreationRequest): Long = petDao.createPet(pet)
+
+    fun getPetsByType(petType: PetType, companyId: Long) = petDao.getPetsByType(petType, companyId)
+
+    fun getPetById(petId: Long, companyId: Long): PetData? = petDao.getPetById(petId, companyId)
+
+    fun getAllPets(companyId: Long) = petDao.getAllPets(companyId)
+
+    fun updatePetOwner(petId: Long, ownerId: Long, companyId: Long) {
+        val success = petDao.updatePetOwner(petId, ownerId, companyId)
+        if(success == 1) return
```

Comment:
Why not just throw exception if it 0? here it is like if..else
2 checks
You can change it to a single check

Reply:
i changed this function in a different branch but because I didn't do the rebase yet it's not updated.
The function is:
    fun updatePetOwner(petId: Long, ownerId: Long, companyId: Long) {
        val success = petDao.updatePetOwner(petId, ownerId, companyId) > 0
        if(success) return
        throw IllegalStateException("Pet doesn't exist or pet already has an owner")
    }

Do you think i should change it to something like this?
    fun updatePetOwner(petId: Long, ownerId: Long, companyId: Long) {
        val success = petDao.updatePetOwner(petId, ownerId, companyId) == 0
        if(success)
              throw IllegalStateException("Pet doesn't exist or pet already has an owner")
    }